libr/anal/block.c

```diff
@@ -543,3 +558,246 @@ R_API R_NULLABLE RList/*<RAnalBlock *>*/ *r_anal_block_shortest_path(RAnalBlock
 	return ret;
 }
 
+typedef struct {
+	RAnalBlock *block;
+	bool reachable;
+} NoreturnSuccessor;
+
+static void noreturn_successor_free(HtUPKv *kv) {
+	NoreturnSuccessor *succ = kv->value;
+	r_anal_block_unref (succ->block);
+	free (succ);
+}
+
+static bool noreturn_successors_cb(RAnalBlock *block, void *user) {
+	HtUP *succs = user;
+	NoreturnSuccessor *succ = R_NEW0 (NoreturnSuccessor);
+	if (!succ) {
+		return false;
+	}
+	r_anal_block_ref (block);
+	succ->block = block;
+	succ->reachable = false; // reset for first iteration
+	ht_up_insert (succs, block->addr, succ);
+	return true;
+}
+
+static bool noreturn_successors_reachable_cb(RAnalBlock *block, void *user) {
+	HtUP *succs = user;
+	NoreturnSuccessor *succ = ht_up_find (succs, block->addr, NULL);
+	if (!succ) {
```

Comment:
be positive, use if (succ) { ... succ->reachable } return true

only one return statement here